SampleGallery/Samples/SDK Insider/Gears/Gears.cs

```diff
@@ -13,126 +13,258 @@
+            };
+
+            // Set the coordinates of where is should be
+            Canvas.SetLeft(image, x);
+            Canvas.SetTop(image, y);
+
+            PerformLayoutCalculation();
+
+            // Add the gear to the container
+            this.Container.Children.Add(image);
+
+            // Add a gear visual to the screen
+            var gearVisual = this.AddGear(image);
+
+            ConfigureGearAnimation(gearVisuals[this.gearVisuals.Count - 1], gearVisuals[this.gearVisuals.Count - 2]);
+        }
+
+        private Visual AddGear(Image gear)
+        {
+            // Create a visual based on the XAML object
+            var visual = ElementCompositionPreview.GetElementVisual(gear);
+            visual.Size = new Vector2((float)gear.ActualWidth, (float)gear.ActualHeight);
+            visual.AnchorPoint = new Vector2(0.5f, 0.5f);
+            this.gearVisuals.Add(visual);
+
+            this.Count++;
+
+            return visual;
+        }
+
+        private void ConfigureGearAnimation(Visual currentGear, Visual previousGear)
+        {
+            // If rotation expression is null then create an expression of a gear rotating the opposite direction
+            rotationExpression = rotationExpression ?? compositor.CreateExpressionAnimation("-previousGear.RotationAngleInDegrees");
+
+            // put in placeholder parameters
+            rotationExpression.SetReferenceParameter("previousGear", previousGear);
+
+            // Start the animation based on the Rotation Angle in Degrees.
+            currentGear.StartAnimation("RotationAngleInDegrees", rotationExpression);
         }
 
         private void EnsureGearMotor()
         {
-            if (_gearMotor == null)
+            // Start the first gear (the red one)
+            if (gearMotionScalarAnimation == null)
             {
-                _gearMotor = _compositor.CreateScalarKeyFrameAnimation();
-                var linear = _compositor.CreateLinearEasingFunction();
+                gearMotionScalarAnimation = compositor.CreateScalarKeyFrameAnimation();
+                var linear = compositor.CreateLinearEasingFunction();
 
-                _gearMotor.InsertExpressionKeyFrame(0.0f, "this.StartingValue");
-                _gearMotor.InsertExpressionKeyFrame(1.0f, "this.StartingValue + 360", linear);
+                gearMotionScalarAnimation.InsertExpressionKeyFrame(0.0f, "this.StartingValue");
+                gearMotionScalarAnimation.InsertExpressionKeyFrame(1.0f, "this.StartingValue + 360", linear);
 
-                _gearMotor.IterationBehavior = AnimationIterationBehavior.Forever;
+                gearMotionScalarAnimation.IterationBehavior = AnimationIterationBehavior.Forever;
             }
         }
 
-        private void AnimateSlow_Click(object sender, RoutedEventArgs e)
+        private void AnimateFast_Click(object sender, RoutedEventArgs e)
         {
+            // Start red gear
```

Comment:
this function doesn't "Start" the gear - the comment should be "// Setup the animation on the red gear" (or similar)